platform/fabric/core/generic/committer/committer.go

```diff
@@ -51,12 +53,13 @@ type committer struct {
 
 	quietNotifier bool
 
-	listeners      map[string][]chan TxEvent
-	mutex          sync.Mutex
-	pollingTimeout time.Duration
+	listeners       map[string][]chan TxEvent
+	mutex           sync.Mutex
+	pollingTimeout  time.Duration
+	serviceProvider *view.ServiceProvider
 }
 
-func New(channel string, network Network, finality Finality, waitForEventTimeout time.Duration, quiet bool, metrics Metrics) (*committer, error) {
+func New(channel string, network Network, finality Finality, waitForEventTimeout time.Duration, quiet bool, metrics Metrics, sp *view.ServiceProvider) (*committer, error) {
```

Comment:
sp is used to get the subscribed, can we pass directly an instance of subscriber?